x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/BaseDateTimeProcessor.java

```diff
@@ -42,12 +44,19 @@ public Object process(Object input) {
             return null;
         }
 
-        if (!(input instanceof ZonedDateTime)) {
-            throw new SqlIllegalArgumentException("A date is required; received {}", input);
+        if (input instanceof ZonedDateTime) {
+            return doProcess(((ZonedDateTime) input).withZoneSameInstant(zoneId));
+        }
+        if (input instanceof OffsetTime) {
+            return doProcess(((OffsetTime) input).withOffsetSameInstant(ZoneOffset.UTC));
         }
 
-        return doProcess(((ZonedDateTime) input).withZoneSameInstant(zoneId));
+        throw new SqlIllegalArgumentException("A [date], a [time] or a [datetime] is required; received {}", input);
     }
 
     abstract Object doProcess(ZonedDateTime dateTime);
-}
\ No newline at end of file
+
+    Object doProcess(OffsetTime time) {
```

Comment:
Same as above - why go through the hassle of creating a time object if only to discard it in the `doProcess`.